metaflow/datastore/task_datastore.py

```diff
@@ -173,6 +173,26 @@ def __init__(
                 if data_obj is not None:
                     self._objects = data_obj.get("objects", {})
                     self._info = data_obj.get("info", {})
+        elif self._mode == "d":
+            self._objects = {}
+            self._info = {}
+
+            if self._attempt is None:
+                for i in range(metaflow_config.MAX_ATTEMPTS):
+                    check_meta = self._metadata_name_for_attempt(
+                        self.METADATA_ATTEMPT_SUFFIX, i
+                    )
+                    if self.has_metadata(check_meta, add_attempt=False):
+                        self._attempt = i
+
+            # Do not allow destructive operations on the datastore if attempt is still in flight
+            # and we explicitly did not allow operating on running tasks.
+            if not allow_not_done and not self.has_metadata(self.METADATA_DONE_SUFFIX):
```

Comment:
is there a case where we want to delete logs for running tasks? It seems in delete mode we may want to ingore `allow_not_done`?

Reply:
I left this in mainly in order to keep the datastore side more generic (allowing delete for in-flight attempts as well, though no use-case ). It is definitely not required for this feature, so I could remove it